### DISCUSSION TITLE
v7.0.0-beta25 breaks deploy:update_code with user in hostname

### DISCUSSION THREAD

**Discussion by underdpt** [deployphp/deployer#2666] [Help needed]
Hello,

With v7.0.0-beta25, this setup:

```php
set('repository', 'ssh://git@git.server.com:2233/folder/project.git');
set('ssh_multiplexing', PHP_OS_FAMILY === 'Windows' ? false : true);
set('default_timeout', null);

host('my.server.host')
    ->set('labels', ['stage' => 'production'])
    ->set('branch', 'master')
    ->set('remote_user', 'mainsip')
    ->set('port', 2223)
    ->set('deploy_path', '/var/www/project');
```

produces this error:

```
info deploying master
task deploy:setup
task deploy:lock
task deploy:release
task deploy:update_code
[my.server.host]  error  in update_code.php on line 54:
[my.server.host] run ssh-keyscan -p 2233 -H git@git.server.com >> ~/.ssh/known_hosts
[my.server.host] err getaddrinfo git@git.server.com: Name or service not known
[my.server.host] err getaddrinfo git@git.server.com: Name or service not known
[my.server.host] err getaddrinfo git@git.server.com: Name or service not known
[my.server.host] err getaddrinfo git@git.server.com: Name or service not known
[my.server.host] err getaddrinfo git@git.server.com: Name or service not known
[my.server.host] exit code 1 (General error)
```

I think the error might be that the hostname is including the user part `git@...` for the ssh-keyscan command.

This works with v7.0.0-beta24

**Comment by sts-ryan-holton** [comment]
I've got the same issue, any workarounds?

**Reply by underdpt** [reply]
Are you still on beta? I think this was solved, at least I don't have this issue anymore.

**Reply by sts-ryan-holton** [reply]
Nope, I'm on v7 rc3 and the issue persists.

**Reply by antonmedv** [reply]
As a workaround populate known_host yourself.

**Reply by sts-ryan-holton** [reply]
Where abouts? I'm using GitHub aliases on my server if that makes any difference with deploy keys for multiple projects that I plan to upgrade to deployed 7.

Also using ipv6

**Reply by antonmedv** [reply]
Abouts?

**Reply by sts-ryan-holton** [reply]
What do I put into `known_hosts`? The public deploy key?

**Reply by sts-ryan-holton** [reply]
@antonmedv I have an `ed25519` deploy key on my server called something like: `fudge_api_ed25519` and a github alias in my `config` file that looks a bit like:

```
Host fudge-api github.com
Hostname github.com
IdentityFile /root/.ssh/fudge_api_id_ed25519
```

In Deployer 6 I could deploy easily with this set up, but with Deployer 7 rc3 this isn't working. The command it's trying to run which is failing for me is:

```
ssh-keygen -F fudge-api:22 || ssh-keyscan -p 22 -H fudge-api >> ~/.ssh/known_hosts
```

Not sure what to do here to get past this.

**Reply by sts-ryan-holton** [reply]
Ok, tried logging into the server and changing this line of `config` -> `Host fudge-api github.com` to -> `Host fudge-api`, now when I run `ssh-keyscan -p 22 -H fudge-api github.com >> ~/.ssh/known_hosts` alone, it seems to work, but still getting the same error when deploying.

**Reply by antonmedv** [reply]
I will delete ssh-keygen automatic population.
